lib/metrics.dart

```diff
@@ -57,21 +57,25 @@ class Metrics {
   final Future<http.Response> Function(http.Request) poster;
   final Function(String, [dynamic]) emit;
   final DateTime Function() clock;
+  final String connectionId;
+  final String userAgent;
   bool disableMetrics;
   Timer? timer;
   Bucket bucket;
   Uri url;
 
-  Metrics(
-      {required this.appName,
-      required this.metricsInterval,
-      required this.clock,
-      this.disableMetrics = false,
-      required this.poster,
-      required this.url,
-      required this.clientKey,
-      required this.emit})
-      : bucket = Bucket(clock);
+  Metrics({
+    required this.appName,
+    required this.metricsInterval,
+    required this.clock,
+    this.disableMetrics = false,
+    required this.poster,
+    required this.url,
+    required this.clientKey,
+    required this.connectionId,
+    required this.userAgent,
```

Comment:
If this is called `userAgent`, but it's not what we set as `user-agent`, will that be confusing? Should it be `unleashUserAgent` or `xUnleashUserAgent` or something? 🤷🏼

Reply:
renamed to `sdkName`